spec/services/feedex_service_spec.rb

```diff
@@ -0,0 +1,93 @@
+require 'gds_api/support_api'
+RSpec.describe FeedexService do
+  let(:support_api) { double('support_api') }
+  let(:subject) { described_class.new(date, 20, support_api) }
+  describe '#find_in_batches' do
+    let(:date) { Date.new(2018, 2, 1) }
+    let(:response_1) do
+      {
+        'results' => [
+          { 'path' => '/path/1', 'count' => 3 },
+          { 'path' => '/path/2', 'count' => 1 }
+        ],
+        'pages' => 3,
+        'current_page' => 1
+      }
+    end
+    let(:response_2) do
+      {
+        'results' => [
+          { 'path' => '/path/3', 'count' => 5 },
+          { 'path' => '/path/4', 'count' => 2 }
+        ],
+        'pages' => 3,
+        'current_page' => 2
+      }
+    end
+    let(:response_3) do
+      {
+        'results' => [
+          { 'path' => '/path/5', 'count' => 1 },
+          { 'path' => '/path/6', 'count' => 7 }
+        ],
+        'pages' => 3,
+        'current_page' => 3
+      }
+    end
+
+    before :each do
+      allow(support_api).to receive(:feedback_by_day)
+        .with(date, 1, 20).and_return(response_1)
+      allow(support_api).to receive(:feedback_by_day)
+        .with(date, 2, 20).and_return(response_2)
+      allow(support_api).to receive(:feedback_by_day)
+        .with(date, 3, 20).and_return(response_3)
+    end
+
+    it 'makes the correct request and yields the first page' do
+      expect { |b| subject.find_in_batches(&b) }.to yield_successive_args(
+        [
+          {
+            date: date,
```

Comment:
It is a bit better to be explicit in the expectation and don't rely on a variable defined within the test.